controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -384,9 +387,24 @@ public CompletableFuture<CreateStreamStatus> createStream(String scope, String s
             return CompletableFuture.completedFuture(
                     CreateStreamStatus.newBuilder().setStatus(CreateStreamStatus.Status.INVALID_STREAM_NAME).build());
         }
+        try {
+            validateStreamConfig(streamConfig);
+        } catch (IllegalArgumentException e) {
+            log.error(requestId, "Create stream: {} failed due to invalid retention policy {}", stream, streamConfig);
+            return CompletableFuture.completedFuture(
+                    CreateStreamStatus.newBuilder().setStatus(CreateStreamStatus.Status.INVALID_RETENTION_POLICY).build());
+        }
```

Comment:
NIT: can we create a method for it, Looks like its a repetitive code (L387-389).